src/main/java/gregicality/science/common/pipelike/pressure/BlockPressurePipe.java

```diff
@@ -0,0 +1,112 @@
+
+    private static final String TRANSLATION_KEY = "gcys.pipe.pressure_pipe_normal";
+
+    public BlockPressurePipe() {
+        setHarvestLevel("pickaxe", 1);
+    }
+
+    @Override
+    protected PressurePipeData createProperties(@Nonnull PressurePipeType pressurePipeType) {
+        return pressurePipeType.modifyProperties(PressurePipeData.EMPTY);
+    }
+
+    @Override
+    public Class<PressurePipeType> getPipeTypeClass() {
+        return PressurePipeType.class;
+    }
+
+    @Override
+    public WorldPressureNet getWorldPipeNet(World world) {
+        return WorldPressureNet.getWorldPipeNet(world);
+    }
+
+    @Override
+    public TileEntityPipeBase<PressurePipeType, PressurePipeData> createNewTileEntity(boolean b) {
+        return new TileEntityPressurePipe();
+    }
+
+    @Override
+    protected PressurePipeData getFallbackType() {
+        return PressurePipeData.EMPTY;
+    }
+
+    @Override
+    public void getSubBlocks(@Nonnull CreativeTabs creativeTabs, @Nonnull NonNullList<ItemStack> items) {
+        for (PressurePipeType type : PressurePipeType.values()) {
+            items.add(new ItemStack(this, 1, type.ordinal()));
+        }
+    }
+
+    @Override
+    public boolean canPipesConnect(IPipeTile<PressurePipeType, PressurePipeData> iPipeTile, EnumFacing enumFacing, IPipeTile<PressurePipeType, PressurePipeData> iPipeTile1) {
+        return iPipeTile1 instanceof TileEntityPressurePipe;
+    }
+
+    @Override
+    public boolean canPipeConnectToBlock(IPipeTile<PressurePipeType, PressurePipeData> iPipeTile, EnumFacing enumFacing, @Nullable TileEntity tileEntity) {
+        return tileEntity != null && tileEntity.hasCapability(GCYSTileCapabilities.CAPABILITY_PRESSURE_CONTAINER, enumFacing);
+    }
+
+    @Override
+    public boolean isHoldingPipe(EntityPlayer player) {
+        if (player == null) {
+            return false;
+        }
+        ItemStack stack = player.getHeldItemMainhand();
+        return !stack.isEmpty() && stack.getItem() instanceof ItemBlockPressurePipe;
+    }
+
+    @Override
+    protected Pair<TextureAtlasSprite, Integer> getParticleTexture(World world, BlockPos blockPos) {
+        return PressurePipeRenderer.INSTANCE.getParticleTexture(getPipeTileEntity(world, blockPos));
+    }
+
+    @SuppressWarnings("deprecation")
+    @Nonnull
+    @Override
+    public EnumBlockRenderType getRenderType(@Nonnull IBlockState state) {
+        return PressurePipeRenderer.INSTANCE.getBlockRenderType();
+    }
+
+    @Nonnull
+    @Override
+    public String getTranslationKey() {
+        return TRANSLATION_KEY;
+    }
+
+    @Override
+    public void breakBlock(@Nonnull World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state) {
+        TileEntity te = worldIn.getTileEntity(pos);
+        if (te instanceof TileEntityPressurePipe) {
+            TaskScheduler.scheduleTask(worldIn, ((TileEntityPressurePipe) te)::updateLeakage);
+        }
+        super.breakBlock(worldIn, pos, state);
```

Comment:
If gt6 pipes is of, the connected pipes will close automatically. Is this intended?